CharlieBackend.Business/Services/HomeworkStudentService.cs

```diff
@@ -159,33 +169,94 @@ public async Task<IList<HomeworkStudentDto>> GetHomeworkStudentForMentor(long ho
             return _mapper.Map<IList<HomeworkStudentDto>>(homeworkStudent);
         }
 
-        private async IAsyncEnumerable<string> ValidateHomeworkStudentRequest(HomeworkStudentRequestDto homeworkStudent, Student student , Homework homework)
+        public async Task<Result<HomeworkStudentDto>> UpdateMarkAsync(UpdateMarkRequestDto request)
         {
-            
-            if (homeworkStudent == default)
+            var homeworkStudent = await _unitOfWork.HomeworkStudentRepository.GetByIdAsync(request.StudentHomeworkId);
+
+            long accountId = _currentUserService.AccountId;
+
+            Mark mark;
```

Comment:
Please remove that and on the line 182 just assign homework Student.Mark = new Mark...